### PR TITLE
[snapshot] update SnapshotHelper.swift to work with tvOS

### LINKED ISSUE #17359: SnapshotHelper.swift fails on tvOS
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When I run my UITests on Apple TV, I get the following error with [SnapshotHelper 1.24](https://github.com/fastlane/fastlane/commit/8ec8fcd1e963a456fe5e0a916a63d733ca2acf94#diff-1a1173200436690d61ef63b6702ae6d6)

```
SnapshotHelper.swift: value of type 'XCUIDevice' has no member 'orientation'
```

The reason is that [UIDeviceOrientation](https://developer.apple.com/documentation/uikit/uideviceorientation) is not supported on tvOS.

The fix is simple, wrap the failing line in `#if os(iOS)` like this:

```
#if os(iOS)
    let image = XCUIDevice.shared.orientation.isLandscape ?  fixLandscapeOrientation(image: screenshot.image) : screenshot.image
#else
    let image = screenshot.image
#endif
```

### PR DESCRIPTION
**Pull request by felginep**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Fix #17359

### Description

Avoid using `UIDeviceOrientation` on tvOS.

### CODE REVIEW

**Comment by felginep** [comment]
@googlebot I signed it!

File: snapshot/lib/assets/SnapshotHelper.swift [hunk 1]
```diff
@@ -294,4 +298,4 @@ private extension CGFloat {
 
 // Please don't remove the lines below
 // They are used to detect outdated configuration files
-// SnapshotHelperVersion [1.24]
+// SnapshotHelperVersion [1.25]
```
Comment (rogerluan) [inline snapshot/lib/assets/SnapshotHelper.swift:301]: This can be reverted as it's auto-increased as part of the release process 🙏

**Review by rogerluan — APPROVED** [review]
Neat fix 🙌  
Nice catch, and thanks for your contribution! ❤️

File: snapshot/lib/assets/SnapshotHelper.swift (hunk 1 above)
Comment (felginep) [reply snapshot/lib/assets/SnapshotHelper.swift]: Done

**Review by joshdholtz — APPROVED** [review]
This is a good catch! Thank you for fixing this ❤️

**Pull request merged**